app/controllers/bookmarks_controller.rb

```diff
@@ -9,7 +9,9 @@ def index
 
   def new
     @bookmark            = Bookmark.new
-    @bookmark.name       = params[:query].to_s.strip.split(/\s| = |!|~|>|</)[0]
+    query                = params[:query].to_s.squish
```

Comment:
let's try s`name = "Some_____very bad name"`/_/ /g - (issues like this is what makes me thing squish is not the best to handle user input)